### PR TITLE
support custom dill params

### PR DESCRIPTION
**Pull request by dmi-feo**
For the sake of https://github.com/ytsaurus/ytsaurus/issues/663

Why `NO_CHOICE` has been introduced:
Since we have some logic for the params (we redefine them for interactive environments), we need to distinguish the situation where the user did not set any value ( == made no choice) with the situation where they actually did (and user's choice should be used no matter what). `None` is a bad option since one day it might become a meaningful value for the params. And `dill_kwargs` cannot be empty by default - `VerifiedDict` has a very strict check for keys when setting (there are a few exceptions in `keys_to_ignore`, but it looks like a crutch, so I guess we don't want to add new keys). Similar approach is used in aiohttp (https://github.com/aio-libs/aiohttp/blob/fc32f69e16f221133d8ee01d2e9928fde35ff02a/aiohttp/client.py#L437), they use enum for such values, but it doesn't seem to make much difference.

### CODE REVIEW

**Comment by Kontakter** [comment]
We already have such a class: https://github.com/ytsaurus/ytsaurus/blob/main/yt/python/yt/wrapper/cypress_commands.py#L25

I offer to generalize it and use it here, by generalize I mean move it in common.py

**Comment by dmi-feo** [comment]
Thank you! I updated the PR

**Review by Kontakter — APPROVED** [review]

File: yt/python/yt/wrapper/tests/test_operations_pickling.py [hunk 1]
```diff
@@ -494,3 +499,25 @@ def func(row):
...
         assert failures <= 1
+
+    @authors("dmifedorov")
+    def test_set_custom_pickler_params(self):
+        @yt_dataclass
+        class OutputRow:
+            str: str
+
+        dynamic_dataclass = yt_dataclass(make_dataclass('Row', fields=[('str', str)]))
+        rows = [dynamic_dataclass(str='a')]
```
[the remark this answers is not in the document]
Comment (dmi-feo) [reply yt/python/yt/wrapper/tests/test_operations_pickling.py]: fixed

File: yt/python/yt/wrapper/default_config.py [hunk 2]
```diff
@@ -12,6 +12,7 @@
 
 import os
 import sys
+import enum
```
[the remark this answers is not in the document]
Comment (dmi-feo) [reply yt/python/yt/wrapper/default_config.py]: Not needed, it was there in early versions, now it is removed

**Review by Kontakter — APPROVED** [review]

**Comment by Kontakter** [comment]
This test is broken, let figure out how to fix it

https://github.com/ytsaurus/ytsaurus/blob/f2ddadc0ad28116e44988d7f5701558d9beb6089/yt/python/yt/wrapper/tests/test_misc.py#L1365

**Review by Kontakter — CHANGES_REQUESTED** [review]
See comment above

**Review by Kontakter — APPROVED** [review]

**Pull request closed without merging**